auth/google.js

```diff
@@ -0,0 +1,60 @@
+import passport from 'passport';
+import { OAuth2Strategy as GoogleStrategy } from 'passport-google-oauth';
+import { User } from '../models';
+
+passport.use(new GoogleStrategy({
+  clientID: process.env.GOOGLE_CLIENT_ID,
+  clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+  callbackURL: 'http://127.0.0.1:7000/api/v1/auth/google/callback',
+},
+/**
+   * callback function for google strategy
+   * @param {object} accessToken authorization token
+   * @param  {object} refreshToken authorization token
+   * @param  {object} profile a user profile
+   * @param {function} done end of function
+   * @returns {function} callback
+   */
+async (accessToken, refreshToken, profile, done) => {
+  const email = profile.emails[0].value;
+  /**
+   * @description - finds an existing user or create a new user
+   * @param {object} user a user
+   * @param {function} done end of function
+   * @returns {object} createOrFindUser
+   */
+  const user = await User.findOrCreate(
+    {
+      where: { email },
+      defaults: {
+        name: profile.displayName,
+        username: profile.name.givenName,
```

Comment:
Here we can destructure the properties of `profile` so we won't keep repeating `profile`.